Modules/_curses_panel.c

```diff
@@ -182,67 +176,87 @@ class _curses_panel.panel "PyCursesPanelObject *" "&PyCursesPanel_Type"
 /*[clinic input]
 _curses_panel.panel.bottom
 
+    cls: defining_class
+
 Push the panel to the bottom of the stack.
 [clinic start generated code]*/
 
 static PyObject *
-_curses_panel_panel_bottom_impl(PyCursesPanelObject *self)
-/*[clinic end generated code: output=7aa7d14d7e1d1ce6 input=b6c920c071b61e2e]*/
+_curses_panel_panel_bottom_impl(PyCursesPanelObject *self, PyTypeObject *cls)
+/*[clinic end generated code: output=8ec7fbbc08554021 input=6b7d2c0578b5a1c4]*/
 {
-    return PyCursesCheckERR(bottom_panel(self->pan), "bottom");
+    _curses_panelstate *state =
+        (_curses_panelstate *)PyType_GetModuleState(cls);
```

Comment:
`(_curses_panelstate *)` cast is useless: PyType_GetModuleState returns `void*` type. You can remove it to fit into a single line. Same comment below.

Reply:
@vstinner fixed.  This shows my ignorance about the C language :)